crates/bevy_color_blindness/src/lib.rs

```diff
@@ -0,0 +1,412 @@
+                [0.0, 0.73333, 0.26667].into(),
+                [0.0, 0.18333, 0.81667].into(),
+            ),
+            Mode::Achromatopsia => (
+                [0.299, 0.587, 0.114].into(),
+                [0.299, 0.587, 0.114].into(),
+                [0.299, 0.587, 0.114].into(),
+            ),
+            Mode::Achromatomaly => (
+                [0.618, 0.32, 0.62].into(),
+                [0.163, 0.775, 0.62].into(),
+                [0.163, 0.320, 0.516].into(),
+            ),
+        }
+    }
+
+    /// Changes `self` to the next Mode.
+    ///
+    /// Useful for writing something like the following:
+    ///
+    /// ```rust
+    /// # use bevy_app::prelude::*;
+    /// # use bevy_color_blindness::*;
+    /// # use bevy_ecs::prelude::*;
+    /// # use bevy_input::prelude::*;
+    /// fn cycle_mode(input: Res<Input<KeyCode>>, mut params: ResMut<ColorBlindnessParams>) {
+    ///     if input.just_pressed(KeyCode::N) {
+    ///         params.mode.cycle();
+    ///         println!("Changed to {:?}", params.mode);
+    ///     }
+    /// }
+    /// ```
+    pub fn cycle(&mut self) {
+        *self = match self {
+            Mode::Normal => Mode::Protanopia,
+            Mode::Protanopia => Mode::Protanomaly,
+            Mode::Protanomaly => Mode::Deuteranopia,
+            Mode::Deuteranopia => Mode::Deuteranomaly,
+            Mode::Deuteranomaly => Mode::Tritanopia,
+            Mode::Tritanopia => Mode::Tritanomaly,
+            Mode::Tritanomaly => Mode::Achromatopsia,
+            Mode::Achromatopsia => Mode::Achromatomaly,
+            Mode::Achromatomaly => Mode::Normal,
+        };
+    }
+}
+
+/// Our custom post processing material
+#[derive(AsBindGroup, TypeUuid, Clone)]
+#[uuid = "bc2f08eb-a0fb-43f1-a908-54871ea597d5"]
+struct PostProcessingMaterial {
+    /// In this example, this image will be the result of the main camera.
+    #[texture(0)]
+    #[sampler(1)]
+    source_image: Handle<Image>,
+
+    #[uniform(2)]
+    red: Vec3,
+    #[uniform(2)]
+    green: Vec3,
+    #[uniform(2)]
+    blue: Vec3,
+}
+
+impl Material2d for PostProcessingMaterial {
+    fn fragment_shader() -> ShaderRef {
+        ShaderRef::Handle(COLOR_BLINDNESS_SHADER_HANDLE.typed())
+    }
+}
+
+/// Component to identify your main camera
+///
+/// Adding this component to a camera will set up the post-processing pipeline
+/// which simulates color blindness. This is done by changing the render target
+/// to be an image, and then using another camera to render that image.
+///
+/// Cameras with `ColorBlindnessCamera` will have [`UiCameraConfig`] inserted with
+/// `show_ui` set to `false`. This is to ensure that UI elements are not rendered twice.
+/// In most cases, you will want to render UI using the final post-processing camera.
+/// If for some reason this behavior is not desired, please open an issue.
+///
+/// [`UiCameraConfig`]: bevy_ui::entity::UiCameraConfig
+#[derive(Component)]
+pub struct ColorBlindnessCamera;
```

Comment:
Hmm. I wonder if this is necessary. It feels like we're controlling the color-blindness display in two different ways, as both a marker component and a resource.

Reply:
Yeah I wasn't too convinced by this either. I'm gonna try to rework it when adding support for multiple windows.